fhevm/tee_pad_gas.go

```diff
@@ -0,0 +1,31 @@
+package fhevm
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func teePadGasRequiredGas(environment EVMEnvironment, suppliedGas uint64, input []byte) uint64 {
+
+	input = input[:minInt(32, len(input))]
+
+	gasLimit := environment.GasLimit()
+	consumedGas := gasLimit - suppliedGas
+
+	padTo := &big.Int{}
+	padTo.SetBytes(input)
+
+	if consumedGas > padTo.Uint64() {
+		return 0
+	} else if padTo.Uint64() > gasLimit {
+		return suppliedGas
```

Comment:
Returning suppliedGas seems incorrect.

```suggestion
```

TBH I would simply remove this line. If padTo > gasLimit, then we padTo that huge amount, and zbc-go-ethereum will take care of throwing an ErrOutOfGas down the line

Reply:
I was thinking that if the user wants to pad to 110 but the gasLimit is 100, then returning suppliedGas will cause the transaction to execute with 100 gas total rather than throwing an error. Wouldn't this be a better user experience since they might not know what the gaslimit is?

Reply:
OK I see what you mean.

If I understand well, there's only one edge case this will happen in the happy path without reverting, it's when `padGas()` is called last, in the last function of a transaction. In all other cases, if there's anything happening after `padGas()`, then the transaction will revert with ErrOutOfGas. Does that seem right?

IMO, if yes, this seems like a small unintuitive thing we need to put in docs. Whereas the obvious way to think of pasGas is to pad until the given amount.

Reply:
Yes this only really happens if ```padGas()``` is called at the end of the function so I see why it would make sense to revert to ErrOutOfGas in all cases just for continuity.